src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "Terminal.hpp"
+
+using namespace Microsoft::Terminal::Core;
+
+// Method Description:
+// - Helper to determine the selected region of the buffer. Used for rendering.
+// Return Value:
+// - A vector of rectangles representing the regions to select, line by line. They are absolute coordinates relative to the buffer origin.
+std::vector<SMALL_RECT> Terminal::_GetSelectionRects() const
+{
+    std::vector<SMALL_RECT> selectionArea;
+
+    if (!_selectionActive)
+    {
+        return selectionArea;
+    }
+
+    // Add anchor offset here to update properly on new buffer output
+    SHORT temp1, temp2;
+    THROW_IF_FAILED(ShortAdd(_selectionAnchor.Y, _selectionAnchor_YOffset, &temp1));
+    THROW_IF_FAILED(ShortAdd(_endSelectionPosition.Y, _endSelectionPosition_YOffset, &temp2));
+
+    // create these new anchors for comparison and rendering
+    const COORD selectionAnchorWithOffset = { _selectionAnchor.X, temp1 };
+    const COORD endSelectionPositionWithOffset = { _endSelectionPosition.X, temp2 };
+
+    // NOTE: (0,0) is top-left so vertical comparison is inverted
+    const COORD &higherCoord = (selectionAnchorWithOffset.Y <= endSelectionPositionWithOffset.Y) ? selectionAnchorWithOffset : endSelectionPositionWithOffset;
+    const COORD &lowerCoord = (selectionAnchorWithOffset.Y > endSelectionPositionWithOffset.Y) ? selectionAnchorWithOffset : endSelectionPositionWithOffset;
+
+    selectionArea.reserve(lowerCoord.Y - higherCoord.Y + 1);
+    for (auto row = higherCoord.Y; row <= lowerCoord.Y; row++)
+    {
+        SMALL_RECT selectionRow;

```

Comment:
Not necessary for this PR,  but in cases like this do you guys recommend initializing these structs (when they can't be initialized at declaration time.
Other projects I work on have a rule where this should be SMALL_RECT{};  I think this is a good rule and prevents unintialized vars.
Just checking for future reviews.  @adiviness

Reply:
I personally prefer initializing any variable with some known value when it's created. I'm sorta hoping there's a static analysis check to enforce this at some point.